pkg/volume/quobyte/quobyte.go

```diff
@@ -0,0 +1,247 @@
+		api.ReadOnlyMany,
+		api.ReadWriteMany,
+	}
+}
+
+func getVolumeSource(spec *volume.Spec) (*api.QuobyteVolumeSource, bool, error) {
+	if spec.Volume != nil && spec.Volume.Quobyte != nil {
+		return spec.Volume.Quobyte, spec.Volume.Quobyte.ReadOnly, nil
+	} else if spec.PersistentVolume != nil &&
+		spec.PersistentVolume.Spec.Quobyte != nil {
+		return spec.PersistentVolume.Spec.Quobyte, spec.ReadOnly, nil
+	}
+
+	return nil, false, fmt.Errorf("Spec does not reference a Quobyte volume type")
+}
+
+func (plugin *quobytePlugin) NewMounter(spec *volume.Spec, pod *api.Pod, _ volume.VolumeOptions) (volume.Mounter, error) {
+	return plugin.newMounterInternal(spec, pod, plugin.host.GetMounter())
+}
+
+func (plugin *quobytePlugin) newMounterInternal(spec *volume.Spec, pod *api.Pod, mounter mount.Interface) (volume.Mounter, error) {
+	source, readOnly, err := getVolumeSource(spec)
+	if err != nil {
+		return nil, err
+	}
+
+	return &quobyteMounter{
+		quobyte: &quobyte{
+			volName: spec.Name(),
+			user:    source.User,
+			group:   source.Group,
+			mounter: mounter,
+			pod:     pod,
+			plugin:  plugin,
+		},
+		registry: source.Registry,
+		volume:   source.Volume,
+		readOnly: readOnly}, nil
+}
+
+func (plugin *quobytePlugin) NewUnmounter(volName string, podUID types.UID) (volume.Unmounter, error) {
+	return plugin.newUnmounterInternal(volName, podUID, plugin.host.GetMounter())
+}
+
+func (plugin *quobytePlugin) newUnmounterInternal(volName string, podUID types.UID, mounter mount.Interface) (volume.Unmounter, error) {
+	return &quobyteUnmounter{&quobyte{
+		volName: volName,
+		mounter: mounter,
+		pod:     &api.Pod{ObjectMeta: api.ObjectMeta{UID: podUID}},
+		plugin:  plugin,
+	}}, nil
+}
+
+// Quobyte volumes represent a bare host directory mount of an quobyte export.
+type quobyte struct {
+	volName string
+	pod     *api.Pod
+	user    string
+	group   string
+	mounter mount.Interface
+	plugin  *quobytePlugin
+	volume.MetricsNil
+}
+
+type quobyteMounter struct {
+	*quobyte
+	registry string
+	volume   string
+	readOnly bool
+}
+
+var _ volume.Mounter = &quobyteMounter{}
+
+func (mounter *quobyteMounter) GetAttributes() volume.Attributes {
+	return volume.Attributes{
+		ReadOnly:        mounter.readOnly,
+		Managed:         false,
+		SupportsSELinux: false,
+	}
+}
+
+// SetUp attaches the disk and bind mounts to the volume path.
+func (mounter *quobyteMounter) SetUp(fsGroup *int64) error {
+	pluginDir := mounter.plugin.host.GetPluginDir(strings.EscapeQualifiedNameForDisk(quobytePluginName))
+	return mounter.SetUpAt(pluginDir, fsGroup)
+}
+
+func (mounter *quobyteMounter) SetUpAt(dir string, fsGroup *int64) error {
+	// Check if Quobyte is already mounted on the host in the Plugin Dir
+	// if so we can use this mountpoint instead of creating a new one
+	// IsLikelyNotMountPoint wouldn't check the mount type
+	if mounted, err := mounter.pluginDirIsMounted(dir); err != nil {
+		return err
+	} else if mounted {
+		return nil
+	}
```

Comment:
We mount all Quobyte Volumes inside the Quobyte Plugin dir and if the directory is already mounted everything is fine